treeherder/services/taskcluster.py

```diff
@@ -0,0 +1,137 @@
+        task_definition = self.queue.task(task_id)
+
+        if actions_json['version'] != 1:
+            raise RuntimeError('Wrong version of actions.json, unable to continue')
+
+        return {
+            'staticActionVariables': actions_json['variables'],
+            'actions': self._filter_relevant_actions(actions_json, task_definition),
+        }
+
+    def _submit(self,
+                action=None,
+                decision_task_id=None,
+                task_id=None,
+                input=None,
+                static_action_variables=None) -> str:
+        context = {
+                "taskGroupId": decision_task_id,
+                "taskId": task_id or None,
+                "input": input,
+         }
+        context.update(static_action_variables)
+
+        if action["kind"] == "task":
+            raise NotImplementedError("Unable to submit actions with 'task' kind.")
+
+        if action["kind"] == "hook":
+            hook_payload = jsone.render(action["hookPayload"], context)
+            hook_id, hook_group_id = action["hookId"], action["hookGroupId"]
+
+            decision_task = self.queue.task(decision_task_id)
+            expansion = self.auth.expandScopes({"scopes": decision_task["scopes"]})
+            expression = f"in-tree:hook-action:{hook_group_id}/{hook_id}"
+
+            if not satisfiesExpression(expansion["scopes"], expression):
+                raise RuntimeError(f"Action is misconfigured: decision task's scopes do not satisfy {expression}")
+
+            result = self.hooks.triggerHook(hook_group_id, hook_id, hook_payload)
+            return result["status"]["taskId"]
+
+    @classmethod
+    def _filter_relevant_actions(cls, actions_json: dict, original_task) -> List:
+        relevant_actions = {}
+
+        for action in actions_json['actions']:
+            action_name = action['name']
+            if action_name in relevant_actions:
+                continue
+
+            no_context_or_task_to_check = (not len(action['context'])) and (not original_task)
+            task_is_in_context = (original_task and original_task.get('tags') and
+                                  cls._task_in_context(action['context'], original_task['tags']))
+
+            if no_context_or_task_to_check or task_is_in_context:
+                relevant_actions[action_name] = action
+
+        return list(relevant_actions.values())
+
+    @staticmethod
+    def _get_action(action_array: list, action_name: str) -> str:
+        try:
+            return [a for a in action_array if a["name"] == action_name][0]
+        except IndexError:
+            available_actions = ", ".join(sorted(
+                {a["name"] for a in action_array}
+            ))
+            raise LookupError(f"{action_name} action is not available for this task.  Available: {available_actions}")
+
+    @classmethod
+    def _task_in_context(cls, tag_set_list, task_tags):
```

Comment:
here you don't define a return type, nor the input types as you do you previous methods